### PR TITLE
Backpack 4: adding costumes and sounds from the backpack

### PR DESCRIPTION
**Pull request by paulkaplan**
backpack-adding-costume-sounds

Based on https://github.com/LLK/scratch-gui/pull/2187, only one commit ahead, so only need to review this commit: 
https://github.com/LLK/scratch-gui/commit/b1f1c3ac0b51405a43e5ea67d7766ab05c0d7f5d

This adds drop handlers to the costumes/sounds tabs to allow them to respond to drop events by either (a) adding the asset directly if it is of the same type (i.e. costumes tab + BACKPACK_COSTUME) or switch to the correct tab and then add it (i.e. costume tab + BACKPACK_SOUND switches to the sounds tab, then adds the sound, and vice versa)

### CODE REVIEW

File: src/containers/sound-tab.jsx [hunk 1]
```diff
@@ -130,6 +133,16 @@ class SoundTab extends React.Component {
...
+        } else if (dropInfo.dragType === DragConstants.BACKPACK_COSTUME) {
+            this.props.onActivateCostumesTab();
+            this.props.vm.addCostume(dropInfo.payload.body, {
+                name: dropInfo.payload.name
+            });
+        } else if (dropInfo.dragType === DragConstants.BACKPACK_SOUND) {
+            this.props.vm.addSound({
+                md5: dropInfo.payload.body,
+                name: dropInfo.payload.name
+            }).then(this.handleNewSound);
```
Comment (rschamp) [inline src/containers/sound-tab.jsx]: There's some repetition going on... any way we can help prevent copy-pasted code here?  Maybe a HOC?

File: src/components/backpack/backpack.css [hunk 2]
```diff
@@ -52,4 +52,5 @@
 .backpack-item {
     min-width: 4rem;
     margin: 0 0.25rem;
+    mix-blend-mode: multiply; /* Make white transparent for thumnbnails */
```
Comment (rschamp) [inline src/components/backpack/backpack.css]: Whoa...

Comment (fsih) [reply src/components/backpack/backpack.css]: Would that break sprites that have white parts?

**Comment by rschamp** [comment]
@paulkaplan I made https://github.com/LLK/scratch-www/pull/1928, so if you want to put this up on scratch-www staging, branch `develop` off to `smoke`. After `smoke` builds successfully, there will be a new scratch-gui npm version on the `latest` tag. So then you should be able to rebuild scratch-www's most recent `develop` build to deploy the latest scratch-gui there.

File: src/components/backpack/backpack.css (hunk 2 above)
Comment (paulkaplan) [reply src/components/backpack/backpack.css]: Yea....

**Pull request merged**